packages/neuron-wallet/src/controllers/wallets.ts

```diff
@@ -87,37 +87,34 @@ class WalletsController {
   }
 
   // TODO: import wallet
-  // public static import = ({
-  //   name,
-  //   password,
-  //   mnemonic,
-  //   keystore,
-  // }: {
-  //   name: string
-  //   password: string
-  //   mnemonic: string
-  //   keystore: string
-  // }): Response<boolean> => {
-  //   // generate wallet for service
-  //   const walletStore = new WalletStore()
-  //   let storedKeystore
-  //   if (mnemonic) {
-  //     storedKeystore = Key.fromMnemonic(mnemonic, true, password).getKeystore()
-  //   } else if (keystore) {
-  //     storedKeystore = Key.fromKeystoreString(keystore, password).getKeystore()
-  //   }
-  //   if (storedKeystore) {
-  //     walletStore.saveWallet(name, storedKeystore)
-  //     return {
-  //       status: ResponseCode.Success,
-  //       result: true,
-  //     }
-  //   }
-  //   return {
-  //     status: ResponseCode.Fail,
-  //     msg: 'Failed to import wallet',
-  //   }
-  // }
+  public static import = ({
+    name,
+    password,
+    mnemonic,
+    keystore,
+  }: {
+    name: string
+    password: string
+    mnemonic: string
+    keystore: string
+  }): Response<String> => {
+    const walletId = WalletsController.service.import({
```

Comment:
service should not be related to business logic, format data for DB in the controller and use `WalletController.service.create` to save it.

Reply:
So `WalletController.service.create` is for saving wallet.I will move `generateKey` from `WalletController.service.create` to controller.

Reply:
How do we tell that `service` is not business logic?

Reply:
That's not very accurate what I've said. I mean in our case, the services just do the work of fetching data from DB or network. Other works like verification, format, should be done in controllers, which precede services.

Reply:
@Keith-CY Makes sense. Thanks for the clarification.